compute/cloud-client/src/main/java/compute/disks/DeleteSnapshotsByFilter.java

```diff
@@ -0,0 +1,91 @@
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class DeleteSnapshotsByFilter {
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // Project ID or project number of the Cloud project you want to use.
+    String projectId = "YOUR_PROJECT_ID";
+
+    // Filter to be applied when looking for snapshots for deletion. Learn more about filters here:
+    // https://cloud.google.com/java/docs/reference/google-cloud-compute/latest/com.google.cloud.compute.v1.ListSnapshotsRequest
+    String filter = "FILTER";
+
+    deleteSnapshotsByFilter(projectId, filter);
+  }
+
+  // Delete a snapshot of a disk.
+  private static void deleteSnapshot(String projectId, String snapshotName)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `snapshotsClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (SnapshotsClient snapshotsClient = SnapshotsClient.create()) {
+
+      Operation operation = snapshotsClient.deleteAsync(projectId, snapshotName)
+          .get(3, TimeUnit.MINUTES);
+
+      if (operation.hasError()) {
+        throw new Error("Snapshot deletion failed!" + operation.getError());
+      }
+      System.out.printf("Snapshot deleted: %s", snapshotName);
+    }
+  }
+
+  // List snapshots from a project.
+  private static ListPagedResponse listSnapshots(String projectId, String filter)
+      throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `snapshotsClient.close()` method on the client to safely
+    // clean up any remaining background resources.
+    try (SnapshotsClient snapshotsClient = SnapshotsClient.create()) {
+
+      // Create the List Snapshot request.
+      ListSnapshotsRequest listSnapshotsRequest = ListSnapshotsRequest.newBuilder()
+          .setProject(projectId)
+          .setFilter(filter)
+          .build();
+
+      return snapshotsClient.list(listSnapshotsRequest);
+    }
+  }
+
+  // Deletes all snapshots in project that meet the filter criteria.
+  public static void deleteSnapshotsByFilter(String projectId, String filter)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+
+    for (Snapshot snapshot : listSnapshots(projectId, filter).iterateAll()) {
+      deleteSnapshot(projectId, snapshot.getName());
+    }
+  }
```

Comment:
I don't think these samples should be written as 3 separate functions - it makes it harder to read and counteracts the advice we give to reuse the client.

Reply:
I replicated the method structure from what we have in python (for consistency). https://github.com/googleapis/python-compute/pull/276

Will talk to the team for reformatting this sample. Thanks.

Reply:
Thanks Maciej.

@kurtisvg I've updated the code sample.